src/cio_websocket.h

```diff
@@ -87,6 +87,18 @@ enum cio_websocket_frame_type {
 
 #define CIO_WEBSOCKET_SMALL_FRAME_SIZE 125
 
+struct cio_websocket_write_job {
+	struct cio_websocket_write_job *next;
+	struct cio_write_buffer *wbh;
+	struct cio_write_buffer websocket_header;
+	uint8_t send_header[14];
+	cio_websocket_write_handler handler;
+	void *handler_context;
+	enum cio_websocket_frame_type frame_type;
+	bool last_frame;
+	cio_buffered_stream_write_handler stream_handler;
+};
+
 struct cio_websocket {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this structure so it has no more than 20 fields, rather than the 31 it currently has. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3AS1820)